lib/server.js

```diff
@@ -104,26 +116,29 @@ exports.create = function create(createOptions) {
       return done(new Error('asking to use react router for rendering, but no routes are provided'));
     }
 
-    var componentInstance;
+    var isRedirect = false;
 
     try {
       if (this.useRouter) {
-        match({ routes:createOptions.routes, location:thing},
+        return match({ routes:createOptions.routes, location:thing},
           function(error, redirectLocation, renderProps) {
           if (redirectLocation) {
             debug('server.js match 302 %s', redirectLocation);
 
+            // perform a client-side redirect, see https://github.com/rackt/history/blob/master/docs/Location.md
+            return done(null, format(TEMPLATE_REDIRECT, redirectLocation.pathname + redirectLocation.search));
+
             //res.redirect(302, redirectLocation.pathname + redirectLocation.search);
           } else if (error) {
             debug('server.js match 500 %s', error);
 
             //res.send(500, error.message);
-          } else if (renderProps === null) {
-            debug('server.js match 404 %s', renderProps);
+          } else if (renderProps) {
```

Comment:
Normalized this against the suggested code at [react-router](https://github.com/rackt/react-router/blob/master/docs/guides/advanced/ServerRendering.md)